tsl/src/compression/compression.c

```diff
@@ -2682,13 +2695,63 @@ process_predicates(Chunk *ch, CompressionSettings *settings, List *predicates, L
 																	 collation,
 																	 opcode,
 																	 arg_value,
-																	 false,	  /* is_null_check */
-																	 false)); /* is_null */
+																	 false, /* is_null_check */
+																	 false, /* is_null */
+																	 false	/* is_array_op */
+																	 ));
 						}
 					}
 				}
 			}
 			break;
+			case T_ScalarArrayOpExpr:
+			{
+				ScalarArrayOpExpr *sa_expr = castNode(ScalarArrayOpExpr, node);
+				if (!ts_extract_expr_args(&sa_expr->xpr, &var, &expr, &opno, &opcode))
+					continue;
+
+				if (!IsA(expr, Const))
+				{
+					expr = (Expr *) estimate_expression_value(&root, (Node *) expr);
+					if (!IsA(expr, Const))
+						continue;
```

Comment:
Let's add some test with volatile function just for coverage.

Reply:
done